mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraApplicableFlagDAO.java

```diff
@@ -75,36 +71,26 @@ private PreparedStatement prepareSelect(Session session) {
                 rowOptional.map(row -> new FlagsExtractor(row).getApplicableFlags()));
     }
 
-    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags flags) {
-        Flags newFlags = new FlagsBuilder().add(flags).build();
-        newFlags.remove(Flag.RECENT);
-        newFlags.remove(Flag.USER);
+    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags oldFlags, Flags newFlags) {
+        ImmutableSet<String> oldUserFlags = ImmutableSet.copyOf(oldFlags.getUserFlags());
+        ImmutableSet<String> newUserFlags = ImmutableSet.copyOf(newFlags.getUserFlags());
+        Sets.SetView<String> addedUserFlags = Sets.difference(newUserFlags, oldUserFlags);
```

Comment:
I think:
```
Sets.SetView<String> addedUserFlags = Sets.difference(newFlags.getUserFlags(), oldFlags.getUserFlags());
```
would be shorter and easier to read. We do not need all those temporary variable